src/pkg/projects/dir.targets

```diff
@@ -70,17 +76,21 @@
       <NonWindowsSymbolFile Include="@(NonWindowsNativeFile -> '%(Identity)$(SymbolFileExtension)')" />
       <ExistingNonWindowsSymbolFile Include="@(NonWindowsSymbolFile)" Condition="Exists('%(Identity)')" />
 
+      <DiscoveredSymbolFile Include="@(ExistingWindowsSymbolFile);@(ExistingNonWindowsSymbolFile)" />
+    </ItemGroup>
+
+    <!-- Remove any duplicates in symbol files found by the above searches. -->
+    <RemoveDuplicates Inputs="@(DiscoveredSymbolFile)">
+      <Output TaskParameter="Filtered" ItemName="FilteredDiscoveredSymbolFile"/>
+    </RemoveDuplicates>
+
+    <ItemGroup>
       <!--
-        These files might be directly redisted, and already in File. Duplicates cause validation
-        errors, so don't add files that are already known.
+        Discovered symbol files might already be in File, without IsSymbolFile set. Make sure we
+        keep the discovered one and apply IsSymbolFile=true.
       -->
-      <DiscoveredSymbolFile
-        Include="@(ExistingWindowsSymbolFile);@(ExistingNonWindowsSymbolFile)"
-        IsSymbolFile="true" />
-      
-      <NormalizedFile Include="$([MSBuild]::NormalizePath('%(File.Identity)'))" />
-
-      <File Include="@(DiscoveredSymbolFile)" Exclude="@(NormalizedFile)" />
+      <File Remove="@(FilteredDiscoveredSymbolFile)" />
+      <File Include="@(FilteredDiscoveredSymbolFile)" IsSymbolFile="true" />
```

Comment:
Making `IsSymbolFile` more reliable has the side effect of removing symbol files from the platform manifest. (`pdb`, `so.dbg`, `dylib.dwarf`, `ni.pdb`.) This seems correct to me because symbol files aren't included in the shared framework, and it looks like it's the original intent:

https://github.com/dotnet/core-setup/blob/f7f4b1b0f0d83f9cffe7218a94de726c59514a77/tools-local/tasks/GenerateFileVersionProps.cs#L57-L60

@ericstj, let me know if this is actually bad. 😄

Reply:
Not bad at all, that sounds good to me.